### PR TITLE
[ONNX] Bump to onnx1.15.0 and ort1.17.0 in CI

### PR DESCRIPTION
**Pull request by liqunfu**

### CODE REVIEW

**Review by thiagocrepaldi — APPROVED** [review]
Overall LGTM. Couple of non-blocking comments only

File: .ci/docker/common/install_onnx.sh [hunk 1]
```diff
@@ -29,9 +29,9 @@ pip_install \
   transformers==4.36.2
 
 pip_install coloredlogs packaging
-retry pip_install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ --no-cache-dir --no-input ort-nightly==1.17.0.dev20231005006
 
-pip_install -i https://test.pypi.org/simple/ onnx==1.15.0rc2
+pip_install onnxruntime==1.17.0
+pip_install onnx==1.15.0
 pip_install onnxscript==0.1.0.dev20240117 --no-deps
```
Comment (thiagocrepaldi) [inline .ci/docker/common/install_onnx.sh:35]: Usually it is a good idea to update `onnxscript` to a newer version to possibly catch issues

**Comment by BowenBao** [comment]
> SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - Failed: Test unexpectedly passed

You can remove the xfail for tests that start passing.

File: test/onnx/test_fx_op_consistency.py [hunk 2]
```diff
@@ -1559,6 +1555,18 @@ def skip_torchlib_forward_compatibility(
             "Reshape", "empty tensor"
         ),
     ),
+    skip(
```
Comment (BowenBao) [inline test/onnx/test_fx_op_consistency.py:1558]: Can we use xfail? Only use skip for non-deterministic errors.

File: test/onnx/test_fx_op_consistency.py [hunk 3]
```diff
@@ -190,6 +190,11 @@ def skip_torchlib_forward_compatibility(
         dtypes=onnx_test_common.COMPLEX_TYPES,
         reason=onnx_test_common.reason_dynamo_does_not_support("Addr", "complex64")
     ),
+    xfail(
+        "all",
+        reason=onnx_test_common.reason_onnx_does_not_support(
+            "Op (ReduceXXX) [ShapeInferenceError] axis must be in [-rank, rank-1]. input rank was 0")
```
Comment (BowenBao) [inline test/onnx/test_fx_op_consistency.py:196]: cc @justinchuby

File: test/onnx/test_fx_op_consistency.py [hunk 4]
```diff
@@ -737,6 +733,11 @@ def skip_torchlib_forward_compatibility(
         "narrow",
         reason=onnx_test_common.reason_dynamo_does_not_support("data-dependent"),
     ),
+    xfail(
+        "native_batch_norm",
+        dtypes=(torch.float16,),
+        reason="fixme: ORT optimizer error: https://github.com/microsoft/onnxruntime/issues/16438",
```
Comment (BowenBao) [inline test/onnx/test_fx_op_consistency.py:739]: The linked issue is already fixed. Probably need to update 'reason' to reflect latest error.

File: test/onnx/test_fx_op_consistency.py [hunk 5]
```diff
@@ -775,6 +776,11 @@ def skip_torchlib_forward_compatibility(
         dtypes=onnx_test_common.INT_TYPES,
         reason=onnx_test_common.reason_onnx_does_not_support("AveragePool", "int"),
     ),
+    xfail(
+        "nn.functional.batch_norm",
+        dtypes=(torch.float16,),
+        reason=onnx_test_common.reason_onnx_does_not_support("Conv1d", "int64"),
```
Comment (BowenBao) [inline test/onnx/test_fx_op_consistency.py:782]: Could this be an ORT regression?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 6]
```diff
@@ -7615,6 +7615,13 @@ def forward(self, embedding_matrix, input, offset, weights):
         self.run_test(model, (embedding_matrix, x, offset, w))
 
     @skipIfUnsupportedMinOpsetVersion(11)
+    @skipIfUnsupportedMaxOpsetVersion(17)
+    @unittest.skip(
+        "This test is broken with ONNXRuntime(17): "
+        "when running with onnxruntime 1.17.0 this test fails with the following error:"
+        "FAIL : Non-zero status code returned while running If node. "
+        "Name:'/If' Status Message: if.cc:253 Compute "
+        "If nodes condition input must have exactly one element")
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:7624]: @thiagocrepaldi need someone to take a look, whether it is export bug or regression in ort.

File: test/onnx/test_fx_op_consistency.py (hunk 2 above)
Comment (BowenBao) [reply test/onnx/test_fx_op_consistency.py:1558]: If it is partially failing some subtests, there are decorators that partially xfails based on test inputs.

File: test/onnx/test_fx_op_consistency.py (hunk 4 above)
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:739]: it still fails with float16: onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : Type Error: Type (tensor(float16)) of output arg (_native_batch_norm_legit_functional_1) of node (_aten__native_batch_norm_training_functional_onnx_1) does not match expected type (tensor(float)).

Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:739]: updated the reason.

File: test/onnx/test_fx_op_consistency.py (hunk 5 above)
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:782]: added error message to reason

File: test/onnx/test_fx_op_consistency.py [hunk 7]
```diff
@@ -1142,9 +1133,13 @@ def skip_torchlib_forward_compatibility(
     ),
     xfail(
         "squeeze",
-        variant_name="multiple",
         reason="fixme: Assertion error: result mismatch",
     ),
+    xfail(
+        "squeeze",
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:1139]: reason updated with the link: https://github.com/microsoft/onnxscript/issues/1264

File: test/onnx/test_fx_op_consistency.py [hunk 8]
```diff
@@ -1177,7 +1172,7 @@ def skip_torchlib_forward_compatibility(
         "tensor_split",
         reason=onnx_test_common.reason_dynamo_does_not_support("data-dependent"),
     ),
-    xfail(
+    skip(
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:1175]: same with "_native_batch_norm_legit", some test fails and some passes. could not find the right filter for them. Will talk to someone for help.

File: test/onnx/test_fx_op_consistency.py [hunk 9]
```diff
@@ -138,7 +138,7 @@ def skip_torchlib_forward_compatibility(
         dtypes=(torch.float16,),
         reason="fixme: Assertion error: result mismatch and type error",
     ),
-    xfail(
+    skip(
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:141]: I used different filters but none allowed the test to pass.

File: test/onnx/test_fx_op_consistency.py [hunk 10]
```diff
@@ -1559,6 +1555,18 @@ def skip_torchlib_forward_compatibility(
             "Reshape", "empty tensor"
         ),
     ),
+    skip(
+        "nn.functional.interpolate",
+        variant_name="linear",
+        matcher=lambda sample: sample.input.dtype in (torch.float16, torch.float32),
+        reason="Mismatched elements with high difference",
+    ),
+    skip(
```
Comment (liqunfu) [inline test/onnx/test_fx_op_consistency.py:1564]: this tests of 4 all failed. however when xfail is used it failed with "Failed: Test unexpectedly passed"

File: test/onnx/test_fx_op_consistency.py (hunk 3 above)
Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:196]: Why is this suddenly popped out on ORT==1.17?

File: test/onnx/test_fx_op_consistency.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (titaiwangms) [reply test/onnx/test_fx_op_consistency.py:1175]: If it's about empty inputs, there are a few cases you can search for.

File: test/onnx/test_fx_op_consistency.py (hunk 4 above)
Comment (thiagocrepaldi) [reply test/onnx/test_fx_op_consistency.py:739]: @BowenBao [scalar_type_analysis](https://github.com/pytorch/pytorch/blob/01abb5af2197cb517fc9b6f241d7dfcc4c66f597/torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp#L49) strikes again?

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 6 above)
Comment (thiagocrepaldi) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:7624]: Let's file a bug an fill in the xfail reason with it

File: test/onnx/test_fx_op_consistency.py (hunk 3 above)
Comment (BowenBao) [reply test/onnx/test_fx_op_consistency.py:196]: need investigation whether it's export bug or ort regression.

File: test/onnx/test_fx_op_consistency.py (hunk 4 above)
Comment (BowenBao) [reply test/onnx/test_fx_op_consistency.py:739]: this is fx export. This is a nasty pytorch legacy problem, the torch cpu/gpu behavior doesn't match.

**Review by thiagocrepaldi — COMMENTED** [review]
@liqunfu there are linting errors on the CI

run `lintrunner -a` and merge the push the fixes 

There are some CI errors for ONNX tests too:

```bash
SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - AssertionError: The values for attribute 'shape' do not match: torch.Size([0]) != torch.Size([0, 0]).
SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - AssertionError: The values for attribute 'shape' do not match: torch.Size([0]) != torch.Size([0, 1]).
SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - AssertionError: The values for attribute 'shape' do not match: torch.Size([0]) != torch.Size([0, 3]).
SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - AssertionError: The values for attribute 'shape' do not match: torch.Size([0]) != torch.Size([0, 5]).
SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - AssertionError: The values for attribute 'shape' do not match: torch.Size([1]) != torch.Size([1, 0]).
SUBFAIL [0.0000s] onnx/test_fx_op_consistency.py::TestOnnxModelOutputConsistency_opset_version_18_model_type_TorchModelType.TORCH_EXPORT_EXPORTEDPROGRAMCPU::test_output_match_linalg_vecdot_cpu_float32 - AssertionError: The values for attribute 'shape' do not match: torch.Size([5]) != torch.Size([5, 0]).
```

**Review by titaiwangms — APPROVED** [review]
So _softmax_backward_data is OS issue..

**Comment by liqunfu** [comment]
> So _softmax_backward_data is OS issue..

yes. that makes testing more difficult :(

File: test/onnx/test_fx_op_consistency.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py]: undone _softmax_backward_data

File: test/onnx/test_fx_op_consistency.py (hunk 3 above)
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py:196]: https://github.com/pytorch/pytorch/issues/119441

File: test/onnx/test_fx_op_consistency.py (hunk 4 above)
Comment (liqunfu) [reply test/onnx/test_fx_op_consistency.py]: https://github.com/microsoft/onnxscript/issues/1269

**Review by thiagocrepaldi — APPROVED** [review]

**Pull request closed without merging**